cmake/kokkos_tribits.cmake

```diff
@@ -227,6 +232,86 @@ MACRO(KOKKOS_PACKAGE_POSTPROCESS)
+   KOKKOS_CONFIG_HEADER( KokkosCore_Config_HeaderSet.in KokkosCore_Config_DeclareBackend.hpp "KOKKOS_DECLARE" "decl/Kokkos_Declare" "${FWD_BACKEND_LIST}")
+   KOKKOS_CONFIG_HEADER( KokkosCore_Config_HeaderSet.in KokkosCore_Config_PostInclude.hpp "KOKKOS_POST_INCLUDE" "Kokkos_Post_Include" "${KOKKOS_BACKEND_POST_INCLUDE_LIST}")
+   SET(_DEFAULT_HOST_MEMSPACE "::Kokkos::HostSpace")
+   KOKKOS_OPTION(DEFAULT_DEVICE_MEMORY_SPACE "" STRING "Override default device memory space")
+   KOKKOS_OPTION(DEFAULT_HOST_MEMORY_SPACE "" STRING "Override default host memory space")
+   KOKKOS_OPTION(DEFAULT_DEVICE_EXECUTION_SPACE "" STRING "Override default device execution space")
+   KOKKOS_OPTION(DEFAULT_HOST_PARALLEL_EXECUTION_SPACE "" STRING "Override default host parallel execution space")
+   IF (NOT Kokkos_DEFAULT_DEVICE_EXECUTION_SPACE STREQUAL "")
+      SET(_DEVICE_PARALLEL ${Kokkos_DEFAULT_DEVICE_EXECUTION_SPACE})
+      MESSAGE(STATUS "Override default device execution space: ${_DEVICE_PARALLEL}")
+      SET(KOKKOS_DEVICE_SPACE_ACTIVE ON)
+   ELSE()
+      IF (_DEVICE_PARALLEL STREQUAL "NoTypeDefined")
+         SET(KOKKOS_DEVICE_SPACE_ACTIVE OFF)
+      ELSE()
+         SET(KOKKOS_DEVICE_SPACE_ACTIVE ON)
+      ENDIF()
+   ENDIF()
+   IF (NOT Kokkos_DEFAULT_HOST_PARALLEL_EXECUTION_SPACE STREQUAL "")
+      SET(_HOST_PARALLEL ${Kokkos_DEFAULT_HOST_PARALLEL_EXECUTION_SPACE})
+      MESSAGE(STATUS "Override default host parallel execution space: ${_HOST_PARALLEL}")
+      SET(KOKKOS_HOSTPARALLEL_SPACE_ACTIVE ON)
+   ELSE()
+      IF (_HOST_PARALLEL STREQUAL "NoTypeDefined")
+         SET(KOKKOS_HOSTPARALLEL_SPACE_ACTIVE OFF)
+      ELSE()
+         SET(KOKKOS_HOSTPARALLEL_SPACE_ACTIVE ON)
+      ENDIF()
+   ENDIF()
+   IF (NOT Kokkos_DEFAULT_DEVICE_MEMORY_SPACE STREQUAL "")
+      SET(_DEFAULT_DEVICE_MEMSPACE ${Kokkos_DEFAULT_DEVICE_MEMORY_SPACE})
+   ENDIF()
+   IF (NOT Kokkos_DEFAULT_HOST_MEMORY_SPACE STREQUAL "")
+      SET(_DEFAULT_HOST_MEMSPACE ${Kokkos_DEFAULT_HOST_MEMORY_SPACE})
+   ENDIF()
+   CONFIGURE_FILE(cmake/Kokkos_Set_Default_Spaces.hpp.in Kokkos_Set_Default_Spaces.hpp @ONLY)
+   #We are ready to configure the header
+   CONFIGURE_FILE(cmake/KokkosCore_config.h.in KokkosCore_config.h @ONLY)
+   FOREACH (PLUGIN ${Kokkos_PLUGIN_PATH})
+      INCLUDE(${PLUGIN}/cmake/ConfigurePlugin.cmake OPTIONAL)
+   ENDFOREACH()
+ENDMACRO()
+
+## KOKKOS_INSTALL_ADDITIONAL_FILES - instruct cmake to install files in target destination.
+##                        Includes generated header files, nvcc_wrapper and
+##                        other files provided through plugins
+MACRO(KOKKOS_INSTALL_ADDITIONAL_FILES)
+   INSTALL(PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/bin/nvcc_wrapper DESTINATION ${CMAKE_INSTALL_BINDIR})
+   INSTALL(FILES "${CMAKE_BINARY_DIR}/KokkosCore_config.h" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
Use `${PROJECT_BINARY_DIR}` instead of `${CMAKE_BINARY_DIR}`. If Kokkos is a subproject, this will clutter the top-level build-directory with Kokkos source files.

Reply:
done